### DISCUSSION TITLE
怎么通过 url 加载 js

### DISCUSSION THREAD

**Discussion by boringplay** [umijs/umi#11598] [Q&A]
例如想要实现 `import('https://somedomain.com/a.js')` ，umi 里应该怎么写？

**Comment by fz6m** [comment]
确保这个 js 是 umd 的情况下，可以配置到 [scripts](https://umijs.org/docs/api/config#scripts) 里然后从全局直接用。

**Reply by boringplay** [reply]
因为要 lazy load，所以想直接用 import('xxx')

**Reply by fz6m** [reply]
这种情况自己写个脚本吧，只需要在按需的地方创建 script 然后插入 html 之后在 script.onload 里监听结束就好了，自己加下加载动画。

**Reply by boringplay** [reply]
想要用原生浏览器支持的 import，但是编译器会报错

**Reply by fz6m** [reply]
如果用原生 `import()` ，需要导入 esm 格式的 script 资源，并且这后续可能会自动加载外网额外的资源，同时只支持现代浏览器，不推荐。

**Reply by boringplay** [reply]
诉求是需要用 import 来实现的，是已有的 CDN 资源，业务场景也是只支持现代浏览器

**Reply by fz6m** [reply]
如需要在 html 使用 esm import ，可以参考 script module 相关资料自行学习。

**Reply by boringplay** [reply]
主要是需要在 js 里 import。写 import('xxx') 就会报错
